src/Components/Publishing/Display/Canvas/CanvasVideo.tsx

```diff
@@ -9,10 +9,31 @@ interface VideoProps {
   campaign: any
   src: any
   onInit?: any
+  tracking?: any
 }
 
 @track()
 export class CanvasVideo extends React.Component<VideoProps, any> {
+  trackDuration = memoize((percentComplete) => {
+    this.props.tracking.trackEvent({
+      action: "Video duration",
+      label: "Display ad video duration",
+      percent_complete: percentComplete,
+      campaign_name: this.props.campaign.name,
+      unit_layout: "canvas_standard"
+    })
+  })
+
+  trackSeconds = memoize((secondsComplete) => {
```

Comment:
`memoize` is used here to prevent multiple tracking calls for the same second interval—e.g. `timeupdate` events will fire for unpredictable 3.1, 3.2, 3.5 intervals b/c of how the HTML5 video API works, so we `Math.floor` that time and use memoize as a way to prevent this function being called multiple times with the same floored time.

Reply:
Very cool